src/mixins/AddSendRequestByMessage.ts

```diff
@@ -665,6 +670,270 @@ export function AddSendRequestByMessage<TBase extends ConnectorTUIBaseConstructo
+          type: "number",
+          name: "max",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+      ])
+
+      const requestItem: FormFieldRequestItemJSON = {
+        "@type": "FormFieldRequestItem",
+        mustBeAccepted: true,
+        title: title,
+        settings: {
+          "@type": "DoubleFormFieldSettings",
+          unit: result.unit,
+          min: result.min,
+          max: result.max,
+        },
+      }
+
+      return requestItem
+    }
+
+    private async createIntegerFormFieldRequestItem(title: string) {
+      const result = await prompts([
+        {
+          message: "[Optional] Enter the name of the unit of the requested integer",
+          type: "text",
+          name: "unit",
+          format: (value) => (value.length > 0 ? value : undefined),
+        },
+        {
+          message: "[Optional] Enter an integer as lower limit for the requested integer",
+          type: "number",
+          name: "min",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+        {
+          message: "[Optional] Enter an integer as upper limit for the requested integer",
+          type: "number",
+          name: "max",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+      ])
+
+      const requestItem: FormFieldRequestItemJSON = {
+        "@type": "FormFieldRequestItem",
+        mustBeAccepted: true,
+        title: title,
+        settings: {
+          "@type": "IntegerFormFieldSettings",
+          unit: result.unit,
+          min: result.min,
+          max: result.max,
+        },
+      }
+
+      return requestItem
+    }
+
+    private async createRatingFormFieldRequestItem(title: string) {
+      const result = await prompts([
+        {
+          message: "Enter an integer between five and ten as upper limit for the requested rating",
+          type: "number",
+          name: "maxRating",
+        },
+      ])
+
+      const requestItem: FormFieldRequestItemJSON = {
+        "@type": "FormFieldRequestItem",
+        mustBeAccepted: true,
+        title: title,
+        settings: {
+          "@type": "RatingFormFieldSettings",
+          maxRating: result.maxRating,
+        },
+      }
+
+      return requestItem
+    }
+
+    private async createSelectionFormFieldRequestItem(title: string) {
+      const result = await prompts([
+        {
+          message: "Which options can be selected? (comma-separated)",
+          type: "text",
+          name: "options",
+        },
+        {
+          message: "[Optional] Should multiple selection be allowed?",
+          type: "confirm",
+          name: "allowMultipleSelection",
+          initial: false,
+        },
+      ])
+
+      const options = result.options
+        .split(",")
+        .map((option: string) => option.trim())
+        .filter((option: string) => option.length > 0)
```

Comment:
you can put all of this code in the format function of the prompt.

As you will need this function for formatting AND validation you can add a function for that above the prompt:

```
      const parseOptions = (input: string) => input.split(",").map((option: string) => option.trim()).filter((option: string) => option.length > 0)
```